common/flight-rpc/src/dns_resolver.rs

```diff
@@ -42,17 +42,14 @@ pub struct DNSResolver {
     inner: TokioAsyncResolver,
 }
 
-lazy_static! {
-    static ref INSTANCE: Result<Arc<DNSResolver>> = {
-        match TokioAsyncResolver::tokio_from_system_conf() {
-            Err(error) => Result::Err(ErrorCode::DnsParseError(format!(
-                "DNS resolver create error: {}",
-                error
-            ))),
-            Ok(resolver) => Ok(Arc::new(DNSResolver { inner: resolver })),
-        }
-    };
-}
+static INSTANCE: Lazy<Result<Arc<DNSResolver>>> =
```

Comment:
Can we extract an `create_dns_resolver` method?

Reply:
emmm, We already have an `instance()` function. Perhaps it could be merged in without `Lazy`.